app/models/reservation_validations.rb

```diff
@@ -45,6 +45,23 @@ def available
       "full time period requested.\n")
   end
 
+  def check_consecutive
+    return unless reserved? || requested?
+    return unless equipment_model
+    return unless equipment_model.max_checkout_length
+
+    # get all reservations of same user with same equipment model
+    # that are consecutive to this reservation
+    res = Reservation.for_reserver(reserver).for_eq_model(equipment_model)
+          .consecutive_with(self)
+
+    res.each do |r|
+      next unless duration + r.duration > equipment_model.max_checkout_length
```

Comment:
I'll think more about how to handle multiple consecutive reservations, but I'm not convinced that it's feasible to check more than immediately overlapping reservations. Perhaps it is simpler to disallow any overlapping reservations of the same equipment model, and have the error message tell the reserver to simply renew their other reservation instead of making a new one.

Another thing: the way this is being handled now, it doesn't allow for reservers to have multiples of the same model reserved at once, which I think is allowable in some circumstances?